k256/src/arithmetic/affine.rs

```diff
@@ -140,16 +150,22 @@ impl DecompressPoint<Secp256k1> for AffinePoint {
                     beta.normalize().is_odd().ct_eq(&y_is_odd),
                 );
 
-                Self {
-                    x,
-                    y: y.normalize(),
-                    infinity: Choice::from(0),
-                }
+                Self::new(x, y.normalize())
             })
         })
     }
 }
 
+/// Decompaction using Taproot conventions as described in [BIP 340].
+///
+/// [BIP 340]: https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki
+impl DecompactPoint<Secp256k1> for AffinePoint {
+    fn decompact(x_bytes: &FieldBytes) -> CtOption<Self> {
+        // TODO(tarcieri): implement full `lift_x` algorithm as described in BIP 340
```

Comment:
BIP 340 defines the `lift_x` function as follows:

> The function ''lift_x(x)'', where ''x'' is an integer in range ''0..p-1'', returns the point ''P'' for which ''x(P) = x''<ref>
>    Given a candidate X coordinate ''x'' in the range ''0..p-1'', there exist either exactly two or exactly zero valid Y coordinates. If no valid Y coordinate exists, then ''x'' is not a valid X coordinate either, i.e., no point ''P'' exists for which ''x(P) = x''. The valid Y coordinates for a given candidate ''x'' are the square roots of ''c = x<sup>3</sup> + 7 mod p'' and they can be computed as ''y = &plusmn;c<sup>(p+1)/4</sup> mod p'' (see [https://en.wikipedia.org/wiki/Quadratic_residue#Prime_or_prime_power_modulus Quadratic residue]) if they exist, which can be checked by squaring and comparing with ''c''.</ref> and ''has_even_y(P)'', or fails if no such point exists. The function ''lift_x(x)'' is equivalent to the following pseudocode:
> * Let c = x<sup>3</sup> + 7 mod p.
> * Let y = c<sup>(p+1)/4</sup> mod p.
> * Fail if c &ne; y<sup>2</sup> mod p.
> * Return the unique point P such that x(P) = x and y(P) = y if y mod 2 = 0 or y(P) = p-y otherwise.

Unfortunately I'm unable to compute *y = c<sup>(p+1)/4</sup> mod p* as the only support we have for raising a `FieldElement` to a power is `FieldElement::pow2k` which requires the exponent be a power of 2.

cc @fjarri